src/contracts/pods/EigenPodManager.sol

```diff
@@ -243,6 +576,120 @@ contract EigenPodManager is Initializable, OwnableUpgradeable, Pausable, IEigenP
+        }
+    }
+
+    // @notice Reduces the `podOwner`'s shares by `shareAmount` and performs a call to the DelegationManager to ensure delegated shares are also tracked correctly
+    function _removeShares(address podOwner, uint256 shareAmount) internal {
+        require(podOwner != address(0), "EigenPodManager._removeShares: depositor cannot be zero address");
+        require(shareAmount != 0, "EigenPodManager._removeShares: shareAmount should not be zero!");
+
+        uint256 currentPodOwnerShares = podOwnerShares[podOwner];
+        require(shareAmount <= currentPodOwnerShares, "EigenPodManager._removeShares: shareAmount too high");
+
+        unchecked {
+            currentPodOwnerShares = currentPodOwnerShares - shareAmount;
+        }
+
+        podOwnerShares[podOwner] = currentPodOwnerShares;
+
+        // if the podOwner has delegated shares, record a decrease in their delegated shares
+        if (!isInUndelegationLimbo(podOwner)) {
+            uint256[] memory shareAmounts = new uint256[](1);
+            shareAmounts[0] = shareAmount;
+            IStrategy[] memory strategies = new IStrategy[](1);
+            strategies[0] = beaconChainETHStrategy;
+            delegationManager.decreaseDelegatedShares(podOwner, strategies, shareAmounts);
+        }
+    }
+
+    // @notice Changes the `podOwner`'s shares by `sharesDelta` and performs a call to the DelegationManager to ensure delegated shares are also tracked correctly
+    function _recordBeaconChainETHBalanceUpdate(address podOwner, int256 sharesDelta) internal {
+        if (sharesDelta < 0) {
+            // if change in shares is negative, remove the shares
+            _removeShares(podOwner, uint256(-sharesDelta));
+        } else {
+            // if change in shares is positive, add the shares
+            _addShares(podOwner, uint256(sharesDelta));
+        }      
+    }
+
+    /**
+     * @notice This function is called to decrement withdrawableRestakedExecutionLayerGwei when a validator queues a withdrawal.
+     * @param amountWei is the amount of ETH in wei to decrement withdrawableRestakedExecutionLayerGwei by
+     */
+    function _decrementWithdrawableRestakedExecutionLayerGwei(address podOwner, uint256 amountWei) 
+        internal
+    {
+        ownerToPod[podOwner].decrementWithdrawableRestakedExecutionLayerGwei(amountWei);
+    }
+
+    /**
+     * @notice Withdraws ETH from an EigenPod. The ETH must have first been withdrawn from the beacon chain.
+     * @param podOwner The owner of the pod whose balance must be withdrawn.
+     * @param recipient The recipient of the withdrawn ETH.
+     * @param amount The amount of ETH to withdraw.
+     * @dev Callable only by the StrategyManager contract.
+     */
+    function _withdrawRestakedBeaconChainETH(address podOwner, address recipient, uint256 amount)
+        internal onlyWhenNotPaused(PAUSED_WITHDRAW_RESTAKED_ETH)
```

Comment:
I don't think having this modifier here makes sense.
Didn't we have a previous thread about this in another PR already?

Reply:
IIRC the flows that arrive at calling this function already all check `onlyWhenNotPaused(PAUSED_WITHDRAWALS)`

Reply:
Hmm ok, looking at it, slashShares(), slashQueuedWithdrawal(),  doesnt have a explicit pauser.  So what I'll do is add the onlyWhenNotPaused(PAUSED_WITHDRAWALS) to those functions?  I think that makes sense given that those functions act very similarly to normal withdrawals

Reply:
Lol nvm, I see that it was  removed, fixed!